tools/tpm2_createpolicy.c

```diff
@@ -334,9 +386,51 @@ static bool init(int argc, char *argv[], createpolicypcr_ctx *ctx) {
         }
     }
 
+    if(c_flag) {
+        file_load_tpm_context_from_file(ctx->sapi_context, &key_handle, context_key_file);
+    } else {
+        key_handle = 0x81010002;
+    }
+
     return true;
 }
 
+static bool rsa_decrypt_and_save(createpolicypcr_ctx *ctx) {
+
+    TPMT_RSA_DECRYPT inScheme;
+    TPM2B_DATA label;
+    TPM2B_PUBLIC_KEY_RSA message = TPM2B_TYPE_INIT(TPM2B_PUBLIC_KEY_RSA, buffer);
+
+    TPMS_AUTH_COMMAND session_data;
+    TSS2_SYS_CMD_AUTHS sessions_data;
+    TPMS_AUTH_RESPONSE session_data_out;
+    TSS2_SYS_RSP_AUTHS sessions_data_out;
+    TPMS_AUTH_COMMAND *session_data_array[1];
+    TPMS_AUTH_RESPONSE *session_data_out_array[1];
+
+    session_data_array[0] = &session_data;
+    sessions_data.cmdAuths = &session_data_array[0];
+    session_data_out_array[0] = &session_data_out;
+    sessions_data_out.rspAuths = &session_data_out_array[0];
+    sessions_data_out.rspAuthsCount = 1;
+    sessions_data.cmdAuthsCount = 1;
+
+    inScheme.scheme = TPM_ALG_RSAES;
+    label.t.size = 0;
+
+    TPM_RC rval = Tss2_Sys_RSA_Decrypt(ctx->sapi_context, key_handle,
+            &sessions_data, &cipher_text, &inScheme, &label, &message,
+            &sessions_data_out);
+    if (rval != TPM_RC_SUCCESS) {
+        LOG_ERR("rsaDecrypt failed, error code: 0x%x", rval);
+        return false;
+    }
+
+    return files_save_bytes_to_file("output.bin", message.t.buffer,
```

Comment:
The tool shouldn't create any "surprise" files. What is this used for, isn't struct createpolicypcr_ctx private to this tool? Does it need this state on different invocations? Can you explain what this is doing, I don't understand.